app/controllers/export_file_controller.rb

```diff
@@ -22,21 +22,47 @@ def start
     @id = params[:id]
   end
 
-  def export
-    @delim_type = params[:delim_type]
-    if @delim_type == "comma"
-      filename = params[:model] + params[:id] + ".csv"
+  # Find the filename and delimiter
+  def find_delim_filename(delim_type, other_char, suffix = "")
+    if delim_type == "comma"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = ","
-    elsif @delim_type == "space"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "space"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = " "
-    elsif @delim_type == "tab"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "tab"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = "\t"
-    elsif @delim_type == "other"
-      filename = params[:model] + params[:id] + ".csv"
+    elsif delim_type == "other"
+      filename = params[:model] + params[:id] + suffix + ".csv"
       delimiter = other_char
     end
+    return filename, delimiter
```

Comment:
All other code climate issues have been fixed. 
Kindly review it by going through our deployment, the manual testing part talks about how you can test the feature.

http://wiki.expertiza.ncsu.edu/index.php?title=CSC/ECE_517_Spring_2017/oss_E1729